scvi/model/_scanvi.py

```diff
@@ -198,37 +191,31 @@ def _data_loader_cls(self):
     @property
     def history(self):
         """Returns computed metrics during training."""
-        return {
-            "unsupervised_trainer_history": self.unsupervised_history_,
-            "semisupervised_trainer_history": self.semisupervised_history_,
-        }
+        return self._trainer.logger.history
 
     def train(
         self,
-        n_epochs_unsupervised: Optional[int] = None,
-        n_epochs_semisupervised: Optional[int] = None,
+        max_epochs: Optional[int] = None,
+        n_samples_per_label: Optional[float] = None,
         train_size: float = 0.9,
         validation_size: Optional[float] = None,
         batch_size: int = 512,
         use_gpu: Optional[bool] = None,
+        num_workers: int = 4,
         train_base_model: bool = True,
         lr: float = 1e-3,
         n_epochs_kl_warmup: int = 400,
         n_iter_kl_warmup: Optional[int] = None,
         check_val_every_n_epoch: Optional[int] = None,
-        unsupervised_trainer_kwargs: dict = {},
-        semisupervised_trainer_kwargs: dict = {},
-        unsupervised_task_kwargs: dict = {},
-        semisupervised_task_kwargs: dict = {},
+        trainer_kwargs: dict = {},
```

Comment:
these shouldn't be {} by default. Please take a look at the totalVI train method. I exposed certain important things.